src/key_value_store.js

```diff
@@ -235,30 +222,26 @@ export class KeyValueStoreLocal {
 
     setValue(key, value, options = {}) {
         validateSetValueParams(key, value, options);
+        const getPath = fileName => path.resolve(this.localEmulationPath, fileName);
 
         // Make copy of options, don't update what user passed.
         const optionsCopy = Object.assign({}, options);
 
-        const deletePromisesArr = LOCAL_FILE_TYPES.map(({ extension }) => {
-            const filePath = path.resolve(this.localEmulationPath, `${key}.${extension}`);
-
-            return unlinkPromised(filePath)
-                .catch((err) => {
-                    if (err.code !== 'ENOENT') throw err;
-                });
-        });
-
-        const deletePromise = Promise
-            .all(deletePromisesArr);
+        const deletePromise = readdirPromised(this.localEmulationPath)
+            .then((files) => {
+                const regex = new RegExp(`^${key}\\.[a-z0-9]+$`);
```

Comment:
1) You don't want to create same regex here for each file (imagine thousands of files) but few lines above to do it just once.

2) It's same regex as at line 206 so lets move this to some private method to don't do the same job at two places.

Reply:
fair point indeed